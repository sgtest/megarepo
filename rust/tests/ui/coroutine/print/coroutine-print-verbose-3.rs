// compile-flags: -Zverbose

#![feature(coroutines, coroutine_trait)]

fn main() {
    let x = "Type mismatch test";
    let coroutine :() = || {
    //~^ ERROR mismatched types
        yield 1i32;
        return x
    };
}
