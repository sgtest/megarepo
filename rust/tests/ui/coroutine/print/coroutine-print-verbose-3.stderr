error[E0308]: mismatched types
  --> $DIR/coroutine-print-verbose-3.rs:7:25
   |
LL |       let coroutine :() = || {
   |  ____________________--___^
   | |                    |
   | |                    expected due to this
LL | |
LL | |         yield 1i32;
LL | |         return x
LL | |     };
   | |_____^ expected `()`, found coroutine
   |
   = note: expected unit type `()`
              found coroutine `{main::{closure#0} upvar_tys=?4t witness=?6t}`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
